### DISCUSSION TITLE
Custom commands

### DISCUSSION THREAD

**Discussion by dbkerun** [hardisgroupcom/sfdx-hardis#648] [Q&A]
Good day, had followed the steps in the article: https://medium.com/@nicolasvuillamy/customize-vscode-sfdx-hardis-with-your-own-menus-and-plugins-839faf1d129 and managed to create a custom command and have it displayed in the left panel of Hardis. Following the latest update of Hardis the command is not visible anymore, I have even removed it and replaced it with the article example and even this command/s are not also shown.

Another question on the custom commands. Is it possible to launch bash commands or any other type of scripts from custom Hardis commands?

Thank you in advance!

**Comment by nvuillam** [comment]
hi @dbkerun ,

Yes you can run any custom command, no need for them to be sfdx hardis or even sfdx ones :)

I just tested again the example of the article with latest sfdx-hardis & VsCode sfdx-hardis and it works like a charm ^^

Please could you share your `config/.sfdx-hardis.yml` file ? (after removing client related info)
Did you open VsCode on the root of your repository ?
![image](https://github.com/user-attachments/assets/b300261f-e12a-4a1d-acc5-d307c4ca2701)

**Reply by dbkerun** [reply]
Hello @nvuillam, 

Thank you for your prompt reply. Please see attached config/.sfdx-hardis.yml and image from Hardis side panels.
<img width="686" alt="Screenshot 2024-07-17 at 11 46 37" src="https://github.com/user-attachments/assets/86531168-3b98-486f-9625-d3f4d6724d1e">

<img width="1574" alt="Screenshot 2024-07-17 at 12 03 56" src="https://github.com/user-attachments/assets/d4448e6a-8389-4220-9652-f186985aa78b">


With regards
Dimitrios

**Reply by nvuillam** [reply]
That's really strange...

Please can you run the following command and share the output ?

`sfdx hardis:config:get --level project`

And also share your .sfdx-hardis.yml as text and not image ?

Thanks :)

**Comment by dbkerun** [comment]
Received the following error:
ERROR running hardis:config:get:  EACCES: permission denied, open '/Users/xxxxxxxxxxxxx/.sfdx/sfdx.log'

Until Friday (before updating to the new Hardis version - which triggered a CLI warning: not installed correctly - which I solved via uninstalling everything and installing everything again) the action was visible. Following the update and the uninstall and installation from my side I have no access to the command and I receive this error now...


The .sfdx-hardis.yml code is below

customCommandsPosition: first
customCommands:
  - id: custom-menu
    label: Custom commands
    commands:
      - id: generate-manifest-xml
        label: Generate manifest
        icon: file.svg
        tooltip: Generates a manifest package.xml using local sfdx source files
        command: sfdx force:source:manifest:create --sourcepath force-app --manifestname myNewManifest
        helpUrl: https://megalinter.github.io/
      - id: list-all-orgs
        label: List all orgs
        icon: salesforce.svg
        tooltip: List all orgs that has already been authenticated using sfdx
        command: sfdx force:org:list --all
  - id: custom-menu-2
    label: Another custom menu
    commands:
      - id: echo
        label: Echo something
        icon: user.svg
        tooltip: Useless commands just to show that we can use not sfdx commands too
        command: echo "Something"

**Reply by nvuillam** [reply]
Can u try deleting this .log file ? (Sometimes it can be gigabytes big, could be the problem, that u probably  have with all sfdx commands, not just this one ^^)
